clients/web/test/spec/swaggerSpec.js

```diff
@@ -6,10 +6,12 @@ describe('Test the swagger pages', function () {
         runs(function () {
             expect($('li#resource_system.resource .heading h2 a').text()).toBe('system');
         });
-        // There seems to be some delay between the link showing and when swaggerUi actually
-        // binds the event handler. We don't have a good hook into that binding, so we hack
-        // it with a 0.1s delay instead.
-        waits(100);
+        // There seems to be some delay between the link showing and when
+        // swaggerUi actually binds the event handler.  Wait until jquery
+        // reports that the event is bound
+        waitsFor(function () {
+            return $._data($('li#resource_system.resource .heading h2 a')[0], 'events') !== undefined;
```

Comment:
In general, I like to use `.first()` instead of `[0]` to get the first element of a set of selectors, as the former returns another jQuery selector while the latter returns a DOMElement (which often needs to be coerced back to a selector in order to be useful). The exception is obviously when you're using raw DOM APIs that can't operate on selectors.

In this case, it doesn't really matter, since jQuery will just do the appropriate thing with the first argument to `_data`, regardless of its type.

Reply:
Oddly, jQuery requires the DOM element in the `_data` call, so `.first()` doesn't work here.